creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -842,39 +855,51 @@ class SnapbackSM {
    * - Processes all users in chunks
    * - For every user on an unhealthy replica, issues an updateReplicaSet op to cycle them off
    * - For every (primary) user on a healthy secondary replica, issues SyncRequest op to secondary
-   *
-   * @note refer to git history for reference to `processStateMachineOperationOld()`
    */
   async processStateMachineOperation() {
+    /**
+     * Ensure lock available, and acquire it - else error
+     * @notice Bull will not enqueue further jobs if all active spots are full, but this provides added
+     *    assurance, especially if queue concurrency were ever set to > 1
+     */
+    const lockKey = 'stateMachineQueueJobLock'
+    const lockExp = this.snapbackJobInterval * 2
+    if ((await redis.lock.getLock(lockKey)) === true) {
+      this.logError(
+        `[stateMachineQueue] Cannot start new job; lock held by previous job.`
+      )
+      return
+    }
+    await redis.lock.setLock(lockKey, lockExp)
```

Comment:
we can make this step atomic. instead of getLock and then setLock we can call the redis SETNX command https://redis.io/commands/setnx/

Reply:
ah this is beautiful thank you, i will do. there's also a redis lock package, but didnt think that was necessary

Reply:
updated